skfair/metrics/fairness_report.py

```diff
@@ -0,0 +1,57 @@
+from collections import defaultdict
+
+from sklearn.metrics import confusion_matrix
+import numpy as np
+
+def fairness_report(y_true, y_pred, groups, group_names=None):
```

Comment:
This may be a nice to have, and I'm not sure, but @MBrouns @nsorros would it make sense that we also allow the user to receive a pandas dataframe as output?

Reply:
makes sense, actually in the original classification report they have an argument to return a dict so that or pandas makes sense. will add.

Reply:
I've justed checked their function signature. Where possible it makes sense to copy it as much as possible so we make sure that we adhere to their standard. I also see they have `target_names=None, sample_weight=None, digits=2`. For now I think target names would be good to have. `sample_weight` and `digits` can be saved for later.

Reply:
Added `output_dict` and `labels`. In regards to target_names, should we rename group_names to target_names? Target names is slighly different for this function as we do not care so much about the names of the target variables, more so about the group names or names for the sensitive characteristic. As such I am wondering whether we should differentiate the name to make the distinction clear.

Reply:
It may be that we care about `target` names though. Let's say I am worried about `gender` group and I have an income prediction usecase. There may be unfairness but also inaccuracy. I may be worried about the `low_income` target and the `high_income` target but not the `medium_income` target.

Reply:
Good point, will add.

Reply:
Actually to clarify what is the intended behaviour here? Is it to filter medium income and calculate fairness metrics on the other two? Or to display them by breaking down the sensitive variable by target name, a multi group of some sort where you see the metrics per gender per income_group?

Reply:
I think I had the term `label` mixed up with `target`. But there is a discussion to be had on something that is very related.

This is the current output of two calls to the summary. 

```
       RECALL        FPR       PPVR        FDR        ACC         F1    Support

b       0.667      0.000      0.667      0.000      2.000      0.667          3
r       1.000      0.000      1.000      0.000      2.000      1.000          2


       RECALL        FPR       PPVR        FDR        ACC         F1    Support

b       0.500      0.000      0.333      0.000      0.333      0.400          3
r       1.000      0.000      1.000      0.000      1.000      1.000          4
```

Is it clear what the difference is? This was the call;

```python
y_true = np.array([0, 0, 1, 1, 1, 2, 2])
y_pred = np.array([0, 0, 0, 1, 1, 2, 0])
groups = np.array([1, 0, 1, 0, 1, 0, 1])
group_names = ["b", "r", "b", "r", "r", "r", "b"]

report = classification_fairness_report(
  y_true, y_pred, 
  groups, group_names, labels=[0, 2]
)
print(report)
report = classification_fairness_report(
  y_true, y_pred, 
  groups, group_names, labels=[0, 1]
)
print(report)
```

It would be nice to incorporate the labels into the report somehow.

Reply:
@MBrouns am very curious to hear your insights here. I'm wondering what table layout is going to be the "most obvious".

Reply:
I think the default sklearn implementation has a row per target right? Considering we want it also per group, would a 'nested' table like a pandas multi-index  make sense?

Reply:
@MBrouns This was indeed the direction that I was wondering about. Scikit-learn only concerns itself with reporting per class, since we've added a group for fairness it may also be valid to think on what the most obvious way is to represent both in this summary. 

I'm guessing adding a column/index makes sense. Since the group is not manditory it may be valid to consider it secondary? So first index the label value and the second one the group? Then again ... if the group is given it may be cosidered more important? It's not 100% clear to me what the best behavior is here.

Reply:
I feel like in general i'd be more interested in comparing different groups for a single outcome instead of comparing the same group for different outcomes. That'd mean that putting the label first would make more sense

Reply:
I agree with @MBrouns, I would also be more keen to compare groups easily thus putting label first.

Again to clarify, what is the intended behaviour when labels=None? Do we print the labels and groups or compute the metrics across all labels?

Reply:
It feels to me that being as explicit as possible is the safest here. We wouldn't want the wrong decision being made and by showing both the label and the group the user can check if they understand the table. By computing the metric across all labels, a user might make a wrong assumption.